### PR TITLE
refine SegmentRead streams

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #5900

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/DataStreams/UnorderedTransformAction.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <DataStreams/UnorderedTransformAction.h>
+
+namespace DB
+{
+Block UnorderedTransformAction::getHeader() const
+{
+    return header;
+}
+
+bool UnorderedTransformAction::transform(Block & block)
```
Comment (JaySon-Huang) [inline dbms/src/DataStreams/UnorderedTransformAction.cpp:24]: can we reuse this code for `DMSegmentThreadInputStream`?
https://github.com/pingcap/tiflash/blob/dc194135de55b3b88c348215d24d7f173e9f055b/dbms/src/Storages/DeltaMerge/DMSegmentThreadInputStream.h#L111-L121

**Review by SeaRise — APPROVED** [review]
lgtm

**Review by JaySon-Huang — APPROVED** [review]

**Comment by JaySon-Huang** [comment]
@ywqzzy please update the branch and get it merged

**Comment by ywqzzy** [comment]
> @ywqzzy please update the branch and get it merged

Because the fork repo is in group ywqdev, so I can't modify the config.
I will use my own account in future development.

**Pull request merged**